.github/workflows/size-pr.yml

```diff
@@ -0,0 +1,45 @@
+# https://securitylab.github.com/resources/github-actions-preventing-pwn-requests/
+#
+# Do a build
+# Measure assets sizes
+# Upload artifact
+# Consumed by size.yml for comparison
+name: "Size: PR"
+
+# read-only repo token
+# no access to secrets
+on:
+  pull_request:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+      - uses: ./.github/actions/setup
+      - run: pnpm build
+      - run: sudo snap install dust
+
+      - name: Save PR number
+        run: |
+          mkdir -p ./pr
+          echo ${{ github.event.number }} > ./pr/NR
+
+      - name: "Get sizes for development outputs"
+        id: dev
+        run: |
+          cd dist
+          dust --ignore_hidden \
+              --reverse --apparent-size \
+              --filter ".+.js$" \
+              --no-percent-bars \
+              --only-file \
+              --full-paths > pr/out.txt
+
+
+      - uses: actions/upload-artifact@v4
+        with:
+          name: pr
```

Comment:
I don't think this works -- I think the name is global for all runs and so this will overwrite the artifact for all PRs

Reply:
I think you want to concat the run id

Reply:
done, thanks!